### DISCUSSION TITLE
Redundant upgrade checks for an `upgrade` segment if the version is the latest

### DISCUSSION THREAD

**Discussion by lewis-yeung** [JanDeDobbeleer/oh-my-posh#5825] [General]
Since v24, the upgrade check is always truly performed for an `upgrade` segment, so if OMP is on the latest version, the segment will be disabled and will not be cached, thus causing redundant upgrade checks and increasing the latency significantly when spawning new prompts.

Related code:

https://github.com/JanDeDobbeleer/oh-my-posh/blob/472bd6fba709b5fab74c1783f091f8468cabaefe/src/segments/upgrade.go#L26-L38

https://github.com/JanDeDobbeleer/oh-my-posh/blob/472bd6fba709b5fab74c1783f091f8468cabaefe/src/config/segment.go#L127-L142

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung the fix is to add the `cache` setting by default, I'll add that to the migration script and update the docs. Good catch.

**Reply by lewis-yeung** [reply]
Confirmed to be fixed in v24.0.9. Thanks.
